File: project_euler/016 - Power digit sum/16.c
#include <stdio.h>

main ()
{
    int i = 1, h = 0, sum = 0, a[1000] = {0};

    a[0] = 1;

    for (i = 1; i <= 1000; i++)
    {
        int j;
        int rem = 0;
        int p = 1;

        for (j = 0;  j <= h; j++)
        {
            p = (a[j]*2) + rem;
            a[j] = p % 10;
            rem = p / 10;

            if ( ( rem != 0 ) && ( j == h ) )
                h++;
        }
    }

    for (i = 0; i <= h; i++)
        sum += a[i];

    printf("Sum of the digits of the number 2^1000 = %d\n", sum);

}

File: project_euler/016 - Power digit sum/16.java
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.List;

class A {

    static List<String> l = new ArrayList<String>();
    static int in = 0;
    static BigInteger i = BigInteger.ONE;

    public static void main(String[] args) {

        BigInteger a = new BigInteger("2");
        System.out.println(a);

        for (int j = 1; j < 1001; j++) {
            i = i.multiply(a);
            
            //System.out.println(i);
        }

        System.out.println("2^1000 = " + i);

        String p = i + "";

        for(String r : p.split("", p.length()+1)) {
            l.add(r);
        }

        for (int j = 1; j < l.size(); j++) {
            in = in + Integer.parseInt(l.get(j));
            System.out.println(l.get(j));
        }

        System.out.println("Sum of the digits in the number 2^1000 = " + in); //1366

    }
}